proxy/logging/RolledLogDeleter.h

```diff
@@ -201,14 +201,27 @@ class RolledLogDeleter
    */
   void clear_candidates();
 
+private:
+  /** Sort all the assembled candidates for each LogDeletingInfo.
+   *
+   * After any additions to the @a deleting_info, this should be called before
+   * calling @a take_next_candidate_to_delete because the latter depends upon
+   * the candidate entries being sorted.
+   */
+  void sort_candidates();
+
 private:
   /** The owning references to the set of LogDeletingInfo added to the below
    * hash map. */
-  std::list<std::unique_ptr<LogDeletingInfo>> deletingInfoList;
+  std::deque<std::unique_ptr<LogDeletingInfo>> deletingInfoList;
 
   /** The set of candidates for deletion keyed by log_type. */
   IntrusiveHashMap<LogDeletingInfoDescriptor> deleting_info;
 
   /** The number of tracked candidates. */
   size_t num_candidates = 0;
+
+  /** Whether the candidates have been sorted since the last addition to the
```

Comment:
Isn't that backwards? If the candidates have been sorted since the last addition, why would `candidates_require_sorting` be true?

Reply:
True. I'll update the comment.